backend/inspirehep/pidstore/api/base.py

```diff
@@ -142,3 +142,16 @@ def get_pid_from_record_uri(uri):
         except IndexError:
             return None
         return pid_type, pid_value
+
+    @staticmethod
+    def get_pid_type_from_recid(recid):
+        return (
+            PersistentIdentifier.query.with_entities(PersistentIdentifier.pid_type)
+            .filter_by(pid_value=recid)
+            .one()[0]
```

Comment:
one() should return the object not an array, also it would be better to catch the exception in case it doesn't exists

Reply:
But with_entities returns a tuple. 
Should I? I think it's better to do it in the view.

Reply:
you can use `.scalar()` which unpacks the element.